writer.h

```diff
@@ -46,4 +79,323 @@ class voxel_writer {
 	}
 };
 
+
+class hdf_writer :public abstract_writer {
+
+public:
+	void Write(const std::string& fnc, std::string statement="default") {
+
+		abstract_chunked_voxel_storage* storage = (abstract_chunked_voxel_storage*)voxels_;
+
+		int continuous_count = 0;
+		int planar_count = 0;
+		int constant_count = 0;
+
+		hsize_t nchunks_x = storage->num_chunks().get(0);
+		hsize_t nchunks_y = storage->num_chunks().get(1);
+		hsize_t nchunks_z = storage->num_chunks().get(2);
+
+		BEGIN_LOOP(size_t(0), nchunks_x, 0U, nchunks_y, 0U, nchunks_z)
+			auto c = storage->get_chunk(ijk);
+
+		if (c == nullptr) {
+			constant_count++;
+
+		}
+
+		else {
+
+			if (c->is_explicit()) {
+				continuous_count++;
+			}
+
+			else {
+				if (c->is_constant()) {
+					constant_count++;
+				}
+
+
+				else {
+					planar_count++;
+				
+				}
+			}
+
+
+		}
+
+		END_LOOP;
+
+
+
+		//TODO: Handle existing file (with group) case
+		const H5std_string FILE_NAME(fnc);
+		H5::H5File file(FILE_NAME, H5F_ACC_RDWR);
+		
+		H5::Group operation_group = H5::Group(file.createGroup(statement));
+
+		
+		// Continuous chunks dataset
+		const H5std_string CONTINUOUS_DATASET_NAME("continuous_chunks");
+
+		const hsize_t      NC = continuous_count;
+		const hsize_t      NX = storage->chunk_size();
+		const hsize_t      NY = storage->chunk_size();
+		const hsize_t	   NZ = storage->chunk_size();
+
+		const int      RANK = 4;
+
+		hsize_t     dimsf[4];
```

Comment:
I find all these variable declarations not immensely useful. I know they do it in the samples, but for me it creates an extra indirection between NX and storage->chunk_size(), I'd rather have a more meaningful name for the variables so that we understand the intent.

I'd also prefer the syntax:

    hsize_t dimsf[4] = {continuous_count, chunk_size, chunk_size, chunk_size };

And please remove all the unnecessary spaces between type and variable name. If any of them remain after the reorganization.

    hsize_t	   NZ
           ^^^^^^

`dimsf[RANK]` for consistency

With this advise most of the lines can be eliminated making it much more clear for the casual reader what's going on.